plugins/tiddlywiki/markdown/wrapper.js

```diff
@@ -12,78 +12,142 @@ Wraps up the markdown-js parser for use in TiddlyWiki5
+		var j = findTagWithType(nodes, currentIndex + 1, closingType);
+		out.push({
+			type: 'element',
+			tag: elementTag,
+			children: convertNodes(nodes.slice(currentIndex + 1, j))
+		});
+		return j;
+	}
+
+	for (var i = 0; i < remarkableTree.length; i++) {
+		var currentNode = remarkableTree[i];
+		if (currentNode.type == 'paragraph_open') {
+			i = wrappedElement('p', i, 'paragraph_close', remarkableTree);
+		} else if (currentNode.type == 'heading_open') {
+			i = wrappedElement('h' + currentNode.hLevel, i, 'heading_close', remarkableTree);
+		} else if (currentNode.type == 'bullet_list_open') {
+			i = wrappedElement('ul', i, 'bullet_list_close', remarkableTree);
+		} else if (currentNode.type == 'list_item_open') {
+			i = wrappedElement('li', i, 'list_item_close', remarkableTree);
+		} else if (currentNode.type == 'link_open') {
+			var j = findTagWithType(remarkableTree, i + 1, 'link_close');
+
+			if (currentNode.href.substr(0, 4) == 'http') {
+				// External link
+				out.push({
+					type: 'element',
+					tag: 'a',
+					attributes: {
+						href: { type: 'string', value: currentNode.href }
+					},
+					children: convertNodes(remarkableTree.slice(i + 1, j))
+				});
+			} else {
+				// Internal link
+				out.push({
+				type: 'link',
+				attributes: {
+					to: {type: 'string', value: currentNode.href}
+				},
+				children: convertNodes(remarkableTree.slice(i + 1, j))
+				});
 			}
-		}
-		// Convert internal links to proper wikilinks
-		if (widget.tag === "a" && widget.attributes.href.value[0] === "#") {
-			widget.type = "link";
-			widget.attributes.to = widget.attributes.href;
-			if (widget.attributes.to.type === "string") {
-				//Remove '#' before conversion to wikilink
-				widget.attributes.to.value = widget.attributes.to.value.substr(1);
+			i = j;
+		} else if (currentNode.type.substr(currentNode.type.length - 5) == '_open') {
+			var tagName = currentNode.type.substr(0, currentNode.type.length - 5);
+			i = wrappedElement(tagName, i, tagName + '_close', remarkableTree);
+			} else if (currentNode.type == 'code') {
+			out.push({
+				type: 'element',
+				tag: currentNode.block ? 'pre' : 'code',
+				children: [{type: 'text', text: currentNode.content}]
+			});
+		} else if (currentNode.type == 'fence') {
+			out.push({
+				type: 'codeblock',
+				attributes: {
+				language: { type: 'string', value: currentNode.params },
+				code: { type: 'string', value: currentNode.content }
+				}
+			});
+		} else if (currentNode.type == 'image') {
+			out.push({
+				type: 'image',
+				attributes: {
+				tooltip: { type: 'string', value: currentNode.alt },
+				source: { type: 'string', value: currentNode.src }
+				}
+			});
+		} else if (currentNode.type == 'inline') {
+			out = out.concat(convertNodes(currentNode.children));
+		} else if (currentNode.type == 'text') {
+			let wikiParser = $tw.wiki.parseText('text/vnd.tiddlywiki', currentNode.content, {
+				parseAsInline: false
```

Comment:
I notice that wikitext is parsed in block mode. Might that cause difficulties with it e.g. recognising `!` at the start of a string as a heading?

It may also be worth prepending a pragma to the parse string to restrict the parse rules that are applied e.g.

```
\rules only html image macrocallinline syslink transcludeinline wikilink filteredtranscludeblock macrocallblock transcludeblock
```

Reply:
Added as a config option now